contracts/root/RootChain.sol

```diff
@@ -161,19 +165,42 @@ contract RootChain is Ownable, WithdrawManager {
     stakeManager = StakeManager(_stakeManager);
   }
 
-  function submitHeaderBlock(bytes32 root, uint256 end, bytes sigs) public {
-    uint256 start = currentChildBlock();
+  function getSha256(bytes input) public view returns (bytes20) {
+    bytes32 hash = sha256(input);
+    return bytes20(hash);
+  }
+
+  function submitHeaderBlock(bytes vote, bytes sigs, bytes extradata) public {
+
+    RLP.RLPItem[] memory dataList = vote.toRLPItem().toList();
+    require(keccak256(dataList[0].toData()) == chain, "Chain ID not same");
+    require(keccak256(dataList[1].toData()) == keccak256(roundType), "Round Type Not Same ");
+    require(keccak256(dataList[5].toData()) == keccak256(voteType), "Vote Type Not Same");
```

Comment:
```suggestion
    require(dataList[5].toByte() == voteType, "Vote Type Not Same");
```

Need to check if this will work or note.

Reply:
voteType is at `dataList[4].toByte()` not `dataList[5]`.

Add check for proposer from `dataList[5].toAddress()`. Check if sender is proposer.

Reply:
Check `voteType`/`roundType` check

<img width="766" alt="screen shot 2018-10-29 at 11 36 35 pm" src="https://user-images.githubusercontent.com/1692593/47670585-90e04780-dbd3-11e8-8a6d-ca066b0f6e53.png">